sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/DeserializerBuildHelper.scala

```diff
@@ -44,25 +44,38 @@ object DeserializerBuildHelper {
     upCastToExpectedType(newPath, dataType, walkedTypePath)
   }
 
+  /**
+   * Returns an expression that can be used to deserialize input expression.
+   *
+   * @param expr The input expression that can be used to extract serialized value.
+   * @param nullable Whether deserialized expression evalutes to null value.
+   * @param walkedTypePath The paths from top to bottom to access current field when deserializing.
+   * @param funcForCreatingDeserializer Given input expression and typed path, this function
+   *                                    returns deserializer expression.
+   */
   def deserializerForWithNullSafety(
```

Comment:
we can remove it, according to https://github.com/apache/spark/pull/23908#discussion_r261047155

Reply:
Removed `deserializerForWithNullSafety`? So we call `deserializerForWithNullSafetyAndUpcast` for all cases?

Reply:
have you read the comment? It's the same as `expressionWithNullSafety`

Reply:
yes, but looks like to remove `expressionWithNullSafety` and incorporate it into `deserializerForWithNullSafety`?

Reply:
oh, i see. I don't see the all change around the change to `deserializerForWithNullSafet` there...

Reply:
I think `expressionWithNullSafety` is more general naming so might be preferred one, but `deserializerForWithNullSafety` is also a good name cause we have relevant method `deserializerForWithNullSafetyAndUpcast`. 

So that's a matter of preference and either can be removed. Which method would we prefer to keep?

Reply:
Moving my comment to #23908 since it will keep updated.